OWML.ModHelper.Menus/ModInputCombinationMenu.cs

```diff
@@ -0,0 +1,205 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using OWML.Common;
+using OWML.Common.Menus;
+using OWML.ModHelper.Events;
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace OWML.ModHelper.Menus
+{
+    public class ModInputCombinationMenu : ModPopupMenu, IModInputCombinationMenu
+    {
+        public event Action<string> OnConfirm;
+        public event Action OnCancel;
+
+        public List<IModInputCombinationElement> CombinationElements { get; private set; }
+
+        private List<Selectable> _selectables;
+
+        public string Combination
+        {
+            get
+            {
+                string result = "";
+                for (int i = 0; i < CombinationElements.Count; i++)
+                {
+                    while (i < CombinationElements.Count && CombinationElements[i].Title == "")
+                    {
+                        CombinationElements[i].DestroySelf();
+                    }
+                }
+                for (int i = 0; i < CombinationElements.Count; i++)
+                {
+                    if (CombinationElements[i].Title != "")
+                    {
+                        result += CombinationElements[i].Title;
+                        if (i < CombinationElements.Count - 1)
+                        {
+                            result += "/";
+                        }
+                    }
+                }
+                return result;
+            }
+            set
+            {
+                _selectables = new List<Selectable>();
+                CombinationElements.ForEach(element => element.Destroy());
+                CombinationElements.Clear();
+                foreach (var combination in value.Split('/'))
+                {
+                    AddCombinationElement(combination);
+                }
+                SelectFirst();
+                UpdateNavigation(_selectables);
```

Comment:
Extract to method

Reply:
Why though?

Reply:
Getters and setters should be small/simple. If you need lots of code in there, it suggests maybe getters and setters aren't the best choice. But calling a method from inside get/set makes it seem simple again ;)

Reply:
now it's FillMenu and GenerateCombination